packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.m

```diff
@@ -323,6 +473,52 @@ - (void)testRefreshReceiptRequest {
   [self waitForExpectations:@[ expectation ] timeout:5];
 }
 
+- (void)testRefreshReceiptRequestWithParams {
+  NSDictionary *properties = @{
+    @"isExpired" : @NO,
+    @"isRevoked" : @NO,
+    @"isVolumePurchase" : @NO,
+  };
+
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"completion handler successfully called"];
+  [self.plugin refreshReceiptReceiptProperties:properties
+                                    completion:^(FlutterError *_Nullable error) {
+                                      [expectation fulfill];
+                                    }];
+  [self waitForExpectations:@[ expectation ] timeout:5];
+}
+
+- (void)testRefreshReceiptRequestWithError {
+  NSDictionary *properties = @{
+    @"isExpired" : @NO,
+    @"isRevoked" : @NO,
+    @"isVolumePurchase" : @NO,
+  };
+
+  id mockHandler = OCMClassMock([FIAPRequestHandler class]);
+  InAppPurchasePlugin *plugin = [[InAppPurchasePlugin alloc]
+      initWithReceiptManager:nil
+              handlerFactory:^FIAPRequestHandler *(SKRequest *request) {
+                return mockHandler;
+              }];
+
+  NSError *recieptError = [NSError errorWithDomain:@"errorDomain"
+                                              code:0
+                                          userInfo:@{NSLocalizedDescriptionKey : @"description"}];
+
+  OCMStub([mockHandler
+      startProductRequestWithCompletionHandler:([OCMArg invokeBlockWithArgs:[NSNull null],
+                                                                            recieptError, nil])]);
+
+  [plugin refreshReceiptReceiptProperties:properties
+                               completion:^(FlutterError *_Nullable error) {
+                                 XCTAssertNotNil(error);
```

Comment:
An expectation here too.

Reply:
omg this is scary cuz I actually didnt replace the handler alloc-ing in the actual function, but the tests were still passing. once i added the expectation, it failed - except the expectation wasnt failing, it was the other asserts. 😱